frontend/packages/console-app/src/components/nodes/NodeDetailsPage.tsx

```diff
@@ -1,40 +1,52 @@
 import * as React from 'react';
+import * as _ from 'lodash';
 import { navFactory } from '@console/internal/components/utils';
 import { PodsPage } from '@console/internal/components/pod';
 import { ResourceEventStream } from '@console/internal/components/events';
+import { NodeKind } from '@console/internal/module/k8s';
 import { DetailsPage } from '@console/internal/components/factory';
 import { nodeStatus } from '../../status/node';
 import NodeDetails from './NodeDetails';
 import NodeTerminal from './NodeTerminal';
 import { menuActions } from './menu-actions';
 import NodeDashboard from './node-dashboard/NodeDashboard';
 
-const { editYaml, events, pods } = navFactory;
+const NodeDetailsPage: React.FC<React.ComponentProps<typeof DetailsPage>> = (props) => {
+  const { editYaml, events, pods } = navFactory;
 
-const pages = [
-  {
-    href: '',
-    name: 'Overview',
-    component: NodeDashboard,
-  },
-  {
-    href: 'details',
-    name: 'Details',
-    component: NodeDetails,
-  },
-  editYaml(),
-  pods(({ obj }) => (
-    <PodsPage showTitle={false} fieldSelector={`spec.nodeName=${obj.metadata.name}`} />
-  )),
-  events(ResourceEventStream),
-  {
-    href: 'terminal',
-    name: 'Terminal',
-    component: NodeTerminal,
-  },
-];
-const NodeDetailsPage: React.FC<React.ComponentProps<typeof DetailsPage>> = (props) => (
-  <DetailsPage {...props} getResourceStatus={nodeStatus} menuActions={menuActions} pages={pages} />
-);
+  const pagesFor = React.useCallback(
+    (node: NodeKind) =>
+      _.compact([
+        {
+          href: '',
+          name: 'Overview',
+          component: NodeDashboard,
+        },
+        {
+          href: 'details',
+          name: 'Details',
+          component: NodeDetails,
+        },
+        editYaml(),
+        pods(({ obj }) => (
+          <PodsPage showTitle={false} fieldSelector={`spec.nodeName=${obj.metadata.name}`} />
+        )),
+        events(ResourceEventStream),
+        node?.metadata?.labels?.['node.openshift.io/os_id'] !== 'Windows'
+          ? { href: 'terminal', name: 'Terminal', component: NodeTerminal }
+          : null,
+      ]),
+    [editYaml, events, pods],
```

Comment:
Just confirming these references don't ever change triggering unnecessary evaluations.

Reply:
So this was automatically added by linter upon committing the changes

Reply:
> So this was automatically added by linter upon committing the changes

That doesn't always mean it's right. I just want to check that the references don't change or it will constantly reevaluate.

Reply:
@spadgett so based on my understanding the all three functions should not change, only their parameters can change if the Node gets changed. But then we ned to re-render anyway. Right?